### PR TITLE
Add -o flag to delete-space

### PR DESCRIPTION
**Pull request by hyenaspots**
Command `delete-space -o <org> <space>` deletes the space in the org specified. No org need be targeted to use the command with this flag.

### CODE REVIEW

**Comment by dkoper** [comment]
Hi @hyenaspots 

Why this enhancement? Wouldn't you generally target the space and list its resources (apps, etc.) to be double sure you're not deleting anything important?
(and even if you don't care about double-checking, in certain environments, do we want to make it easy for other users to shoot themselves in the foot?)

Regards,
Dies Koper
CF CLI PM

**Comment by JesseTAlford** [comment]
This change offers three principal improvements.

1. Scriptability. The ability to fully explicate the target of a space deletion in a command is a readability and complexity improvement when using the cli in scripts.
1. Consistency. The delete command should be consistent with the create command, which currently offers this option. Bringing more consistency to the interface makes it easier to discover - and to remember.
1. The Admin experience. Admins of large instances tend not to use the standard login and target flows, which do not serve them well. 

I'll also say that as a developer of CF, I often want to quickly do things in various orgs/spaces in support of testing and troubleshooting, and I'd generally like to make the CLI a stronger tool for this - using 'target' constantly is irritating when I know my environment well.

I disagree that being able to fully-explicate the target of operations is especially concerning in terms of making it easy for users to do themselves harm. If anything, leaving the target implicit/based on state seems the riskier - but, given that we're not revving major versions here (or maybe ever?), I'll be satisfied to fix up this limiting inconsistency.

**Comment by hyenaspots** [comment]
@dkoper Incidentally, I have signed the CLA and received confirmation from Chip Childers. I don't know why cfdreddbot doesn't think I have.

Can we get this merged? The longer it stays open, the greater the chance of a conflict developing.

File: cf/commands/space/delete_space.go [hunk 1]
```diff
@@ -28,12 +30,13 @@ func init() {
 func (cmd *DeleteSpace) MetaData() commandregistry.CommandMetadata {
 	fs := make(map[string]flags.FlagSet)
 	fs["f"] = &flags.BoolFlag{ShortName: "f", Usage: T("Force deletion without confirmation")}
+	fs["o"] = &flags.StringFlag{ShortName: "o", Usage: T("Delete space within specifid org")}
```
Comment (dkoper) [inline cf/commands/space/delete_space.go]: "specified"

File: cf/commands/space/delete_space.go [hunk 2]
```diff
@@ -28,12 +30,13 @@ func init() {
...
 
 	return commandregistry.CommandMetadata{
 		Name:        "delete-space",
 		Description: T("Delete a space"),
 		Usage: []string{
-			T("CF_NAME delete-space SPACE [-f]"),
+			T("CF_NAME delete-space SPACE [-o] [-f]"),
```
Comment (dkoper) [inline cf/commands/space/delete_space.go]: `[-o ORG]` to clarify this option takes a value.

**Comment by dkoper** [comment]
@JesseTAlford Thanks for those details.
@hyenaspots I have commented on some typos.
We're super busy finishing the last stories to get a release out at the moment and prefer not to risk/change the release plan for this enhancement. I'll look at prioritizing it soon after the release before we make any other code changes. Sounds fair?

**Comment by chipchilders** [comment]
closing / opening to re-check CLA

**Comment by chipchilders** [comment]
closing / opening to re-check CLA

**Comment by chipchilders** [comment]
once more with feeling

**Comment by chipchilders** [comment]
once more with feeling

**Comment by chipchilders** [comment]
@dkoper - This can be merged as is. The toolsmiths team can reproduce the bug in the bot... sorry for the confusion @hyenaspots !

**Pull request merged**